Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/Klarna/Managers/KlarnaHeadlessManager.swift

```diff
@@ -0,0 +1,184 @@
+        
+        /// A delegate to handle errors that may occur during the payment process, involving KlarnaHeadlessManager logic.
+        public weak var errorDelegate: PrimerHeadlessErrorableDelegate?
+        
+        /// Component responsible for managing session creation stages of the Klarna payment session.
+        var sessionCreationComponent: KlarnaPaymentSessionCreationComponent?
+        
+        /// Component responsible for managing session authorization stages of the Klarna payment session.
+        var sessionAuthorizationComponent: KlarnaPaymentSessionAuthorizationComponent?
+        
+        /// Component responsible for managing session finalization stages of the Klarna payment session.
+        var sessionFinalizationComponent: KlarnaPaymentSessionFinalizationComponent?
+        
+        /// Component responsible for managing klarna view stages of the Klarna payment session.
+        var viewHandlingComponent: KlarnaPaymentViewHandlingComponent?
+        
+        // MARK: - Init
+        public override init() {
+            super.init()
+            
+            guard let tokenizationComponentProtocol = PrimerAPIConfiguration.paymentMethodConfigTokenizationComponent.first else {
+                let error = PrimerError.unknown(
+                    userInfo: [
+                        "file": #file,
+                        "class": "\(Self.self)",
+                        "function": #function,
+                        "line": "\(#line)"
+                    ],
+                    diagnosticsId: UUID().uuidString)
+                ErrorHandler.handle(error: error)
+                return
+            }
+            self.tokenizationComponent = tokenizationComponentProtocol
+            
+            self.sessionCreationComponent = KlarnaPaymentSessionCreationComponent(
+                tokenizationComponent: tokenizationComponentProtocol
+            )
+            
+            self.sessionAuthorizationComponent = KlarnaPaymentSessionAuthorizationComponent(
+                tokenizationComponent: tokenizationComponentProtocol
+            )
+            
+            self.sessionFinalizationComponent = KlarnaPaymentSessionFinalizationComponent(
+                tokenizationComponent: tokenizationComponentProtocol
+            )
+            self.viewHandlingComponent = KlarnaPaymentViewHandlingComponent()
+        }
+        
+        /// Sets a delegate to handle various events and errors during the Klarna payment process.
+        public func setDelegate(_ delegate: PrimerHeadlessKlarnaComponent) {
+            errorDelegate = delegate
+            validate()
+        }
+        
+        // MARK: - Session creation public methods
+        
+        /// Configures delegates for the session creation component to handle validation, errors, and steps in the payment process.
+        public func setSessionCreationDelegates(_ delegate: PrimerHeadlessKlarnaComponent) {
```

Comment:
Im not sure I agree with this approach, I think we should be hiding the sessionCreationComponent from the user, but this method name is leaking that a little.

A couple of points:
- I think it is ok to have one `errorDelegate` - it can be propagated from the manager to any inner components.
- I would be ok with the merchant setting the delegates explicitly, optionally supplying a method (with appropriate documentation) which sets all three to the same delegate for convenience.

@semirp Thoughts here, how are delegates handled in Android?

Reply:
We use one step delegate for this as we have just one public facing component.